bip-ephemeralanchors.mediawiki

```diff
@@ -0,0 +1,138 @@
+<pre>
+  BIP: ?
+  Layer: Application
+  Title: Ephemeral Anchors
+  Author: Gregory Sanders <gsanders87@gmail.com>
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-?
+  Status: Draft
+  Type: Informational
+  Created: 2023-01-11
+  License: CC0 1.0 Universal
+  Requires: 431
+</pre>
+
+==Introduction==
+
+===Abstract===
+
+Ephemeral Anchors are a mempool policy carve-out that allows dust UTXOs,
+even 0-value, to be created, provided they are also spent in the mempool.
+
+===Motivation===
+
+Relay dust limits have been in place in most implementations of the Bitcoin
+protocol to discourage the creation of UTXOs that are never spent in the future,
+bloat the UTXO set, and increase the validation burden for validating
+nodes.
+
+With [https://github.com/bitcoin/bips/blob/master/bip-0431.mediawiki TRUC] transactions
+and basic package relay, users can generate and propagate 0-fee transactions provided
+a descendant transaction includes the proper fee for the package. In many cases
+these fees can be directly sourced from a freely spendable output, with the resulting
+change output being decremented by the required fee amount.
+
+In some cases however there are no freely spendable outputs to source this fee value
+from, and in these cases an "anchor" is employed.
+
+[https://github.com/lightning/bolts/blob/master/03-transactions.md#to_local_anchor-and-to_remote_anchor-output-option_anchors LN]
+allows a small amount of contract value to be given to an output to allow network relay
+by passing dust checks, but not as the primary source of fee funds. Instead the child transaction
+spending the anchor is responsible for providing the funds.
+
+In this and similar abstractions it would be cleaner if the anchor itself could be 0-value
+instead of requiring anchors to exceed dust amounts.
+
+===Typical Configurations===
+
+For anchors using TRUC transactions, it's expected that they would take two output script forms:
+
+1. Keyed anchor: A key, possibly shared by multiple privileged parties, is used to encumber the anchor. This could also be `tr()`, `p2wsh()` or any
+output type that allows key material.
+1. Un-keyed anchor: `P2SH(OP_TRUE)` or `P2WSH(OP_TRUE)`, depending on the user's need for lack of txid malleability. Further policy
+extensions could allow output templates such as the output script <code>OP_1 <0x4e73></code> or a bare `OP_TRUE`.
```

Comment:
```suggestion
# Keyed anchor: A key, possibly shared by multiple privileged parties, is used to encumber the anchor. This could also be `tr()`, `p2wsh()` or any output type that allows key material.
# Un-keyed anchor: `P2SH(OP_TRUE)` or `P2WSH(OP_TRUE)`, depending on the user's need for lack of txid malleability. Further policy extensions could allow output templates such as the output script <code>OP_1 <0x4e73></code> or a bare `OP_TRUE`.
```

Reply:
removed this section as it's speaking on script templates rather than the key dust concept